backend/api/Repositories/Mayan/MayanBaseRepository.cs

```diff
@@ -0,0 +1,241 @@
+            {
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Token", authenticationToken);
+            }
+            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue(MediaTypeNames.Application.Json));
+
+            try
+            {
+                HttpResponseMessage response = await client.PutAsync(endpoint, content).ConfigureAwait(true);
+                var result = await ProcessResponse<T>(response);
+                return result;
+            }
+            catch (Exception e)
+            {
+                _logger.LogError("Unexpected exception during put {e}", e);
+                return new ExternalResult<T>()
+                {
+                    Status = ExternalResultStatus.Error,
+                    Message = "Exception during Put",
+                };
+
+            }
+        }
+
+        protected async Task<ExternalResult<string>> DeleteAsync(Uri endpoint, string authenticationToken = null)
+        {
+            using HttpClient client = _httpClientFactory.CreateClient("Pims.Api.Logging");
+            client.DefaultRequestHeaders.Accept.Clear();
+            if (!string.IsNullOrEmpty(authenticationToken))
+            {
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Token", authenticationToken);
+            }
+            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue(MediaTypeNames.Application.Json));
+
+            ExternalResult<string> result = new ExternalResult<string>()
+            {
+                Status = ExternalResultStatus.Error,
+                Payload = endpoint.AbsolutePath,
+            };
+
+            try
+            {
+                HttpResponseMessage response = await client.DeleteAsync(endpoint).ConfigureAwait(true);
+
+                _logger.LogTrace("Response: {response}", response);
+
+                string payload = await response.Content.ReadAsStringAsync().ConfigureAwait(true);
+                result.HttpStatusCode = response.StatusCode;
+                switch (response.StatusCode)
+                {
+                    case HttpStatusCode.OK:
+                        _logger.LogTrace("Response payload: {payload}", payload);
+                        result.Status = ExternalResultStatus.Success;
+                        break;
+                    case HttpStatusCode.NoContent:
+                        result.Status = ExternalResultStatus.Success;
+                        result.Message = "No content was returned from the call";
+                        break;
+                    case HttpStatusCode.Forbidden:
+                        result.Status = ExternalResultStatus.Error;
+                        result.Message = "Request was forbidden";
+                        break;
+                    case HttpStatusCode.BadRequest:
+                        result.Status = ExternalResultStatus.Error;
```

Comment:
question - why not use ProcessResponse<T>() here like the other methods do?

Reply:
The delete response needed different logic from the retrieval (GET) or update/create (post, patch) since it does not return a payload